axelrod/match_generator.py

```diff
@@ -167,3 +167,61 @@ def estimated_size(self):
         """Rough estimate of the number of matches that will be generated."""
         size = self.__len__() * (1. / self.prob_end) * self.repetitions
         return size
+
+class SpatialMatches(RoundRobinMatches):
+
+    def __init__(self, players, turns, game, repetitions, edges):
+
+        player_indices = list(range(len(players)))
+        node_indices = sorted(set([node for edge in edges for node in edge]))
+        if player_indices != node_indices:
+            raise ValueError("The graph edges do not include all players.")
+
+        self.edges = edges
+        super(SpatialMatches, self).__init__(players, turns, game, repetitions)
+        """
+        A class that generates spatial structure matches. In spatial structure
+        matches the players do not play against every other player, instead they
+        only play their neighbors.
+
+        In literature the authors interpreted spatial tournaments topology
+        as a 2D square lattice. We want to create a class that will not take as
+        an argument only a 2D lattice but any given graph.
+
+        After considering various python packages for graphs, we ended up using
+        a simple dictionary that includes the edges.So the players - nodes will
+        only play players that are connected to with an edge.
+
+        We have chosen to implement spatial structure tournaments because
```

Comment:
See the following references:
* Nowak and May (1992) <http://www.nature.com/nature/journal/v359/n6398/abs/359826a0.html
* Lindgren and Nordahl (1994) <http://www.sciencedirect.com/science/article/pii/0167278994902895>
* Brauchli, Killingback and Doebeli (1999) <http://www.zoology.ubc.ca/~doebeli/reprints/Doe38.pdf>

If you want to include more information about the literature and history I'd suggest putting that in the documentation instead.